Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -966,8 +978,41 @@ extension NIOAsyncWriter {
                     }
                 } else {
                     // Yield hasn't been marked as cancelled.
-                    // This means we can either call the delegate or suspend
-                    return .init(isWritable: isWritable, delegate: delegate)
+
+                    if isWritable && !inDelegateOutcall {
+                        self._state = .streaming(
+                            isWritable: isWritable,
+                            inDelegateOutcall: true, // We are now making a call to the delegate
+                            cancelledYields: cancelledYields,
+                            suspendedYields: suspendedYields,
+                            elements: elements,
+                            delegate: delegate
+                        )
+
+                        return .callDidYield(delegate)
```

Comment:
What happens to `sequence` in this branch? It looks like we just ignore it

Reply:
I see where the confusion is coming from. We are just yielding it to the delegate right away when we return `callDidYield`. The caller of this method still has the sequence.

Reply:
Ah, gotcha, thanks for clarifying.